libvcx/src/libindy/proofs/proof_request.rs

```diff
@@ -39,26 +39,31 @@ impl ProofRequestData {
     }
 
     pub fn set_requested_attributes(mut self, requested_attrs: String) -> VcxResult<ProofRequestData> {
-        let requested_attributes: Vec<AttrInfo> = ::serde_json::from_str(&requested_attrs)
-            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Invalid Requested Attributes: {:?}, err: {:?}", requested_attrs, err)))?;
-
-        for attribute in requested_attributes.iter() {
-            if attribute.name.is_some() && attribute.names.is_some() {
-                return Err(VcxError::from_msg(VcxErrorKind::InvalidProofRequest,
-                                              format!("Requested attribute can contain either 'name' or 'names'. Not both.")));
-            };
+        match serde_json::from_str::<HashMap<String, AttrInfo>>(&requested_attrs) {
```

Comment:
I like the new format, please add information about the alternate format to `vcx_proof_create`

Reply:
Just let's add at least 1 unit test for the new format